### ISSUE TITLE
Paths in "#include"-statements in headers

### ISSUE THREAD

**Issue by SkyToGround** [ess-dmsc/h5cpp#45]
The header files in ``src/include/h5cpp/`` include other headers either by relative path or by assuming that ``include/h5cpp/`` is an include directory. This should probably changed such that paths are always absolute and that the root include path is ``include`` and not ``include/h5cpp``. This is the method used by Boost and many other libraries as it prevents unintentionally including the wrong header file.

Please comment, I can make the relevant changes.

**Comment by martukas** [comment]
I agree that this could be done cleaner.
It's also the case that the unit tests also reference the library includes via relative paths. It should probably somehow test it in a way that reflects it's use as a library that has it's headers packaged in a cmake variable?

**Comment by SkyToGround** [comment]
I am working on the unit test problem which is why I discovered that this was the case.

**Comment by eugenwintersberger** [comment]
If we do this let me please first finish the IO stuff (should be finished today). Once this is done I can deal with this. As this would require changing a lot of files I would like to do this in one big sweep in order to avoid big merges with a lot of conflicts with all other branches.

**Comment by SkyToGround** [comment]
It turns out that the header inclusion problem I had was caused by something else (which was my own fault). However, let me do a few tests before we close or take action on this issue.

**Comment by SkyToGround** [comment]
So the original reason for doing this does not exist as mentioned in my previous post. However, there might still be a reason to do it as it will clarify the "#include" statements somewhat.

**Comment by eugenwintersberger** [comment]
I had a look on the boost header files and have to admint that the suggested solutions looks quite 
nice. However, I would not rate this as a minor issue but rather as something we could postpone until we have a first alpha release.

**Comment by eugenwintersberger** [comment]
I fixed this along with #112. So it will be closed with the pull request for this issue once it is done.
